analyzers/src/SonarAnalyzer.Common/Rules/CollectionEmptinessCheckingBase.cs

```diff
@@ -0,0 +1,89 @@
+        where TSyntaxKind : struct
+    {
+        private const string DiagnosticId = "S1155";
+
+        protected CollectionEmptinessCheckingBase() : base(DiagnosticId) { }
+
+        protected override void Initialize(SonarAnalysisContext context) =>
+            context.RegisterSyntaxNodeActionInNonGenerated(Language.GeneratedCodeRecognizer,
+                c =>
+                {
+                    var binaryLeft = Language.Syntax.BinaryExpressionLeft(c.Node);
+                    var binaryRight = Language.Syntax.BinaryExpressionRight(c.Node);
+
+                    if (Language.ExpressionNumericConverter.TryGetConstantIntValue(binaryLeft, out var left))
+                    {
+                        CheckExpression(c, binaryRight, left, Language.Syntax.ComparisonKind(c.Node).Mirror());
+                    }
+                    else if (Language.ExpressionNumericConverter.TryGetConstantIntValue(binaryRight, out var right))
+                    {
+                        CheckExpression(c, binaryLeft, right, Language.Syntax.ComparisonKind(c.Node));
+                    }
+                },
+                Language.SyntaxKind.ComparisonKinds);
+
+        private void CheckExpression(SyntaxNodeAnalysisContext context, SyntaxNode expression, int constant, ComparisonKind comparison)
+        {
+            if (comparison.Compare(constant).IsEmptyOrNotEmpty()
+                && TryGetCountCall(expression, context.SemanticModel, out var location, out var typeArgument))
+            {
+                context.ReportIssue(Diagnostic.Create(Rule, location, typeArgument));
+            }
+        }
+
+        private bool TryGetCountCall(SyntaxNode expression, SemanticModel semanticModel, out Location countLocation, out string typeArgument)
+        {
+            if (Language.Syntax.NodeIdentifier(expression) is { } identifier
+                && identifier.ValueText == nameof(Enumerable.Count)
+                && (semanticModel.GetSymbolInfo(identifier.Parent.Parent).Symbol is IMethodSymbol methodSymbol)
+                && IsMethodCountExtension(methodSymbol)
+                && methodSymbol.IsExtensionOn(KnownType.System_Collections_Generic_IEnumerable_T)
+                && methodSymbol.ReceiverType is INamedTypeSymbol receiverType)
+            {
+                countLocation = identifier.Parent.GetLocation();
+                typeArgument = (methodSymbol.TypeArguments.Any()
+                    ? methodSymbol.TypeArguments
+                    : receiverType.TypeArguments).Single().ToDisplayString();
+                return true;
+            }
+            else
+            {
+                countLocation = null;
+                typeArgument = null;
+                return false;
+            }
+        }
+
+        private static bool IsMethodCountExtension(IMethodSymbol methodSymbol) =>
+            methodSymbol.IsExtensionMethod
+            && methodSymbol.ReceiverType != null;
```

Comment:
Please remove this method completely, as it doesn't check the `Count` anymore and we can remove the `methodSymbol.ReceiverType != null` part as well, because we now have `methodSymbol.ReceiverType is INamedTypeSymbol receiverType`.

The `methodSymbol.IsExtensionMethod` can be moved above to the main `if` chain

Reply:
IsExtensionMethod can also be removed, because `IsExtensionOn` above does the same thing